src/redux/sagas/colony/colonyCreate.ts

```diff
@@ -507,6 +366,43 @@ function* colonyCreate({
       yield takeFrom(setOneTxRoles.channel, ActionTypes.TRANSACTION_SUCCEEDED);
     }
 
+    /*
+     * Wait for the colony to exist, then navigate to it.
+     */
+    const colonyExistsSubscrition = yield apolloClient
+      .watchQuery<
+        GetDisplayNameByColonyNameQuery,
+        GetDisplayNameByColonyNameQueryVariables
+      >({
+        query: GetDisplayNameByColonyNameDocument,
+        variables: {
+          name: givenColonyName,
+        },
+        pollInterval: 1000,
+      })
+      .subscribe({
+        next: ({ data: { getColonyByName } }) => {
+          const [colony] = getColonyByName?.items ?? [];
+          const { displayName: existingColonyDisplayName } =
+            colony?.metadata ?? {};
+          if (existingColonyDisplayName && navigate) {
```

Comment:
Is checking for colony not being undefined not enough?

Reply:
Yes it is, however I wanted to also check that `navigation` is being passed through to the saga, in the event that UI will get changed

Reply:
@rdig I was referring just to the fact you're checking for the display name property rather than just the metadata or even the colony itself, is there a particular reason?